packages/flutter/lib/src/material/expansion_panels.dart

```diff
@@ -0,0 +1,294 @@
+/// expanded or collapsed. The body of the panel is only visible when it is
+/// expanded.
+///
+/// Expansion panels are used as children for [ExpansionPanelList].
+///
+/// See also:
+///
+///  * [ExpansionPanelList]
+///  * <https://material.google.com/components/expansion-panels.html>
+class ExpansionPanel {
+  /// Creates an expansion panel to be used as a child for [ExpansionPanelList].
+  ExpansionPanel({
+    this.headerBuilder,
+    this.body,
+    this.isExpanded: false
+  }) {
+    assert(this.headerBuilder != null);
+    assert(this.body != null);
+    assert(this.isExpanded != null);
+  }
+
+  /// The widget builder that builds the header part of the expansion panel
+  /// while providing the expansion state of the current panel.
+  ///
+  /// This widget is always visible.
+  final ExpansionPanelHeaderBuilder headerBuilder;
+
+  /// The body of the expansion panel that site below the header.
+  ///
+  /// This widget is visible only when the panel is expanded.
+  final Widget body;
+
+  /// Whether the panel is expanded.
+  ///
+  /// Defaults to false.
+  final bool isExpanded;
+}
+
+/// A Material expansion panel list that lays out its children and animates
+/// expansions.
+///
+/// See also:
+///
+///  * [ExpansionPanel]
+///  * <https://material.google.com/components/expansion-panels.html>
+class ExpansionPanelList extends StatefulWidget {
+  /// Creates an expansion panel list widget. The [expansionCallback] is
+  /// triggered whenever one of the expansion icon is pushed.
+  ExpansionPanelList({
+    Key key,
+    this.children: const <ExpansionPanel>[],
+    this.expansionCallback
+  }) : super(key: key) {
+    assert(this.children != null);
+  }
+
+  /// The children of the expansion panel list. They are layed in a similar
+  /// fashion to [BlockBody].
+  final List<ExpansionPanel> children;
+
+  /// The callback that gets called whenever one of the expansion icons is
+  /// pressed. The arguments passed to the callback are the index of the
+  /// to-be-expanded panel in the list and whether the panel is currently
+  /// expanded or not.
+  final ExpansionPanelCallback expansionCallback;
+
+  @override
+  _ExpansionPanelListState createState() => new _ExpansionPanelListState();
+}
+
+class _ExpansionPanelListState extends State<ExpansionPanelList> {
+  @override
+  Widget build(BuildContext context) {
+    List<MergeableMaterialItem> items = <MergeableMaterialItem>[];
+
+    for (int i = 0; i < config.children.length; i += 1) {
+      if (config.children[i].isExpanded && i != 0 && !config.children[i - 1].isExpanded)
+        items.add(new MaterialGap(key: new ValueKey<int>(i * 2 - 1)));
+
+      Row header = new Row(
+        children: <Widget>[
+          new Flexible(
+            child: new AnimatedContainer(
+              duration: kThemeAnimationDuration,
+              curve: Curves.ease,
+              margin: config.children[i].isExpanded ? const EdgeInsets.symmetric(
```

Comment:
To shorten this expression, maybe define a isExpanded (config.children[i].isExpanded) loop var and a _kExpandedEdgeInsets const?